### PR TITLE
modules/aws/vpc - Allow us to customise master node SSH sources

### PR DESCRIPTION
**Pull request by mrwacky42**
I would like to pass in trusted subnets for ssh access to our master nodes, rather than the entire dirty IPv4 internet.

### CODE REVIEW

**Comment by squat** [comment]
ok to test

**Comment by squat** [comment]
@mrwacky42 can you please run `make docs examples` to fix the static analysis issues?

File: platforms/aws/variables.tf [hunk 1]
```diff
@@ -259,6 +259,12 @@ Example:
 EOF
 }
 
+variable "tectonic_aws_master_ssh_allowed_ips" {
+  type        = "list"
+  default     = ["0.0.0.0"]
+  description = "(optional) This configures trusted subnets for SSH access to the master nodes."
```
Comment (mxinden) [inline platforms/aws/variables.tf]: @mrwacky42 Would you mind adding an `Example` section like e.g. in `tectonic_aws_worker_custom_subnets`?

File: platforms/aws/variables.tf [hunk 2]
```diff
@@ -259,6 +259,12 @@ Example:
 EOF
 }
 
+variable "tectonic_aws_master_ssh_allowed_ips" {
+  type        = "list"
+  default     = ["0.0.0.0"]
```
Comment (mxinden) [inline platforms/aws/variables.tf]: Our tests are failing with:
`* module.vpc.aws_security_group_rule.master_ingress_ssh: "cidr_blocks.0" must contain a valid CIDR, got error parsing: invalid CIDR address: 0.0.0.0`

As a wild guess, shouldn't this be `["0.0.0.0/0"]`?

Comment (mrwacky42) [reply platforms/aws/variables.tf]: Indeed it should

File: platforms/aws/variables.tf (hunk 1 above)
Comment (mrwacky42) [reply platforms/aws/variables.tf]: Done.

**Comment by mrwacky42** [comment]
rebased and `make structure-check`'d

**Comment by mrwacky42** [comment]
Can I get a heck-yeah?

**Comment by enxebre** [comment]
retest this please

**Comment by enxebre** [comment]
lgtm cc @mxinden @squat

**Comment by cpanato** [comment]
@mrwacky42 can you please rebase and squash your commits? thanks!

**Comment by mrwacky42** [comment]
@cpanato - rebased & squashed.

**Comment by cpanato** [comment]
@squat  can you review when you have time? all tests passed

**Comment by enxebre** [comment]
Hey @mrwacky42 thanks a lot for this and apologies for the inactivity. We've been going through big architecture redesign and  the code here will soon be migrating to a new repo where new features requests will be considerated. More detailed information will be shared soon

**Pull request closed without merging**